### PR TITLE
Fix actions dropdown focus timing

### PR DESCRIPTION
**Pull request by PVince81**
Because "@after-show" is still happening too early due to v-tooltip's
use of "requestAnimationFrame" we can't rely on its events to tell us
when the popover is actually visible.

This workaround introduces an additional Popover event "after-show" and
"after-hide" that are based on monitoring the inner "isOpen" property
which is the one toggling the CSS visibility.

The Actions popover logic now uses "after-show" to reliably apply the
focus on the first action element.

Fixes https://github.com/nextcloud/nextcloud-vue/issues/1760
See https://github.com/Akryum/v-tooltip/issues/661 for the upstream issue.

### CODE REVIEW

File: src/components/Popover/Popover.vue [hunk 1]
```diff
@@ -74,6 +75,24 @@ export default {
...
+	mounted() {
+		this.$watch(
+			() => {
+				// required because v-tooltip doesn't provide events
+				// and @show is too early
+				// see https://github.com/Akryum/v-tooltip/issues/661
+				return this.$refs.popover.isOpen
+			},
+		  (val) => {
+			  if (val) {
```
Comment (PVince81) [inline src/components/Popover/Popover.vue:88]: weird indents... fixing...

**Comment by PVince81** [comment]
tldr; fixes keyboard navigation for after opening the menu :smile:

File: src/components/Popover/Popover.vue [hunk 2]
```diff
@@ -74,6 +75,24 @@ export default {
...
+		this.$watch(
+			() => {
+				// required because v-tooltip doesn't provide events
+				// and @show is too early
+				// see https://github.com/Akryum/v-tooltip/issues/661
+				return this.$refs.popover.isOpen
+			},
+			(val) => {
+				if (val) {
+					this.$emit('after-show')
```
Comment (skjnldsv) [inline src/components/Popover/Popover.vue:89]: Please document those events

**Review by skjnldsv — CHANGES_REQUESTED** [review]

**Review by ma12-co — APPROVED** [review]

**Comment by PVince81** [comment]
@skjnldsv docs added

File: src/components/Popover/Popover.vue (hunk 2 above)
Comment (skjnldsv) [reply src/components/Popover/Popover.vue:102]: @PVince81 you can document them for our styleguide directly, no need for a readme :)
https://vue-styleguidist.github.io/docs/Documenting.html#events

**Review by skjnldsv — CHANGES_REQUESTED** [review]

**Comment by PVince81** [comment]
@skjnldsv nice, thanks for the hint. I've reverted the bits in the header and added docs close to `$emit`. Then squashed into the last commit.

Now it looks like this:
image

**Review by skjnldsv — APPROVED** [review]
giphy
